src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -125,16 +125,27 @@ const DateTimeBlock : React.FC<IProps> = ({
   const [displayMinutes, setDisplayMinutes] = useState<string>(getFormattedTime(convertMinutes));
 
   const setDateHours = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
-    const hourRegex  = /^[0-1]{0,1}[0-9]{0,1}$|(^2[0-4])$/;
-    if (!hourRegex.test(value)) {
-      return;
+    let newVal = '0';
+    if(value.length === 1){
+      if(value === '0'){
+        newVal = ('0');
+      } else {
+        newVal = ('0' + value).slice(-2) ;
+      }
+    } else {
+      newVal = value.slice(-2);
+    }
```

Comment:
We should use `padStart` function to pad 0, and this should be handled with keyboard events for specific up & down keys and arrow buttons in UI. If we handle it here then backspacing will also pad 0 and as user I cannot freely input anything without this padding.

Reply:
We can handle 0 padding in the onKeyDown function when user press Up/Down arrows from keyboard, but in order to handle 0 padding onchange of the arrow buttons in UI then it will call the setDateHours onChange function and hence I have handled this 0 padding here and therefore when a user enters manually then also it triggers onChange event (setDateHours function) which pads 0 to the entered value